src/core/dbus-manager.c

```diff
@@ -1807,6 +1807,63 @@ static int method_get_dynamic_users(sd_bus_message *message, void *userdata, sd_
         return sd_bus_send(NULL, reply, NULL);
 }
 
+static int method_restart_needing_restart(sd_bus_message *message, void *userdata, sd_bus_error *error) {
+        Manager *m = userdata;
+        int r;
+
+        assert(message);
+        assert(m);
+
+        r = mac_selinux_access_check(message, "start", error);
+        if (r < 0)
+                return r;
+
+        r = bus_verify_manage_units_async(m, message, error);
+        if (r < 0)
+                return r;
+        if (r == 0)
+                return 1; /* No authorization for now, but the async polkit stuff will call us again when it has it */
+
+        log_debug("Queuing restart jobs…");
+
+        _cleanup_(sd_bus_message_unrefp) sd_bus_message *reply = NULL;
+        r = sd_bus_message_new_method_return(message, &reply);
+        if (r < 0)
+                return r;
+
+        r = sd_bus_message_open_container(reply, 'a', "o");
+        if (r < 0)
+                return r;
+
+        Unit *u;
+        char *k;
+        HASHMAP_FOREACH_KEY(u, k, m->units) {
+                /* ignore aliases */
+                if (u->id != k)
+                        continue;
+
+                if (!u->needs_restart)
+                        continue;
+
+                r = mac_selinux_unit_access_check(
+                                u, message,
+                                job_type_to_access_method(JOB_RESTART),
+                                error);
+                if (r < 0)
+                        return r;
+
+                r = bus_unit_queue_job_one(message, u, JOB_RESTART, JOB_FAIL, 0, reply, error);
+                if (r < 0)
+                        return r;
+        }
+
+        r = sd_bus_message_close_container(reply);
+        if (r < 0)
+                return r;
+
+        return sd_bus_send(NULL, reply, NULL);
```

Comment:
Supressing errors here doesn't seem nice either. In particular this can fail because of permission errors, and I'm pretty sure we should report those. I left it as is for now. But if you have a good idea how to change the code to ignore the right type of errors, please make a suggestion.

Reply:
well, there's also the other option: continue the loop, but then tell the caller that some enqueuing failed, and to check the logs for details. i.e. it's unlikely we can ever propagate all errors to the client via dbus error messages...

i.e. intrdouce a new dbus error: JobsPartiallyEnqueued() and in the log message say "Enqeueing some jobs failed, for details see logs".

Or maybe include the first error message that was encountered, but not all.

Reply:
Done. I made it return the first error and a message:
"Failed to enqueue some jobs, see logs for details: %m".